base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs

```diff
@@ -158,6 +160,18 @@ impl UnconfirmedPool {
         Ok(selected_txs)
     }
 
+    // This will search a Vec<Arc<Transaction>> for duplicate inputs of a tx
+    fn find_duplicate_input(array_of_tx: &Vec<Arc<Transaction>>, tx: &Arc<Transaction>) -> bool {
```

Comment:
Clippy has suggested an improvement here.

Reply:
will have to add a ignore clippy warning here.